File: problem-solving/baekjoon/16395_파스칼의_삼각형.py
# https://www.acmicpc.net/problem/16395
import sys

if __name__ == "__main__":
    n, k = list(map(int, sys.stdin.readline().split()))
    dp = [[0]*n for _ in range(n)]
    for i in range(n):
        for j in range(i+1):
            if j == 0 or j == i:
                dp[i][j] = 1
            else:
                dp[i][j] = dp[i-1][j-1] + dp[i-1][j]
    print(dp[n-1][k-1])

File: problem-solving/baekjoon/16395_파스칼의_삼각형.go
// https://www.acmicpc.net/problem/16395
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	reader := bufio.NewReader(os.Stdin)
	writer := bufio.NewWriter(os.Stdout)
	defer writer.Flush()

	var n, k int
	fmt.Fscanln(reader, &n, &k)

	var dp = make([][]int, n)
	for i := 0; i < n; i++ {
		dp[i] = make([]int, n)
	}
	for i := 0; i < n; i++ {
		for j := 0; j < i+1; j++ {
			if j == 0 || j == i {
				dp[i][j] = 1
			} else {
				dp[i][j] = dp[i-1][j-1] + dp[i-1][j]
			}
		}
	}
	fmt.Fprintln(writer, dp[n-1][k-1])
}
